app/services/mentor_at_school_periods/change_lead_provider.rb

```diff
@@ -0,0 +1,112 @@
+module MentorAtSchoolPeriods
+  class ChangeLeadProvider
+    class LeadProviderNotChangedError < StandardError; end
+
+    attr_reader :mentor_at_school_period,
+                :lead_provider,
+                :author
+
+    include TrainingPeriodSources
+
+    def initialize(mentor_at_school_period:, lead_provider:, author:)
+      @mentor_at_school_period = mentor_at_school_period
+      @lead_provider = lead_provider
+      @author = author
+    end
+
+    def call
```

Comment:
I like `call` but just FYI, there's a push within the team to stop using generic method names like that and use names that are a bit more descriptive.